arm/Microsoft.Compute/availabilitySets/deploy.bicep

```
@@ -33,19 +33,23 @@ param tags object = {}
 @description('Optional. Customer Usage Attribution ID (GUID). This GUID must be previously registered')
 param cuaId string = ''
 
+var proximityPlacementGroup = {
+  id: proximityPlacementGroupId
+}
+
 module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
   name: 'pid-${cuaId}'
   params: {}
 }
 
-resource availabilitySet 'Microsoft.Compute/availabilitySets@2021-04-01' = {
+resource availabilitySet 'Microsoft.Compute/availabilitySets@2021-07-01' = {
   name: name
   location: location
   tags: tags
   properties: {
     platformFaultDomainCount: availabilitySetFaultDomain
     platformUpdateDomainCount: availabilitySetUpdateDomain
-    proximityPlacementGroup: !empty(proximityPlacementGroupId) ? proximityPlacementGroupId : null
+    proximityPlacementGroup: !empty(proximityPlacementGroupId) ? proximityPlacementGroup : null
```

Most of the modules are avoiding `var `for some reason and using directly
```suggestion
    proximityPlacementGroup: !empty(proximityPlacementGroupId) ? {
      id: proximityPlacementGroupId
    }: null
```
For consistency I'd suggest that and to remove the var above, but I don't have a hard opinion on which of the 2 is the best.

---

Fixed